rotkehlchen/api/rest.py

```diff
@@ -3660,7 +3660,22 @@ def get_associated_locations(self) -> Response:
 
     def get_location_labels(self) -> Response:
         with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
-            labels = [x[0] for x in cursor.execute('SELECT DISTINCT location_label FROM history_events')]  # noqa: E501
+            # Get distinct location_labels with their corresponding location
+            # Ordered by frequency (most frequent first)
+            # When multiple locations exist for a label, we take the first one
+            results = cursor.execute(
+                'SELECT location_label, MIN(location) as location, COUNT(*) as frequency '
+                'FROM history_events WHERE location_label IS NOT NULL '
+                'GROUP BY location_label '
+                'ORDER BY frequency DESC',
+            ).fetchall()
+            labels = [
+                {
+                    'location_label': row[0],
+                    'location': Location.deserialize_from_db(row[1]).serialize(),
+                }
+                for row in results
+            ]
```

Comment:
```suggestion
            labels = [{
                'location_label': row[0],
                'location': Location.deserialize_from_db(row[1]).serialize(),
            } for row in results]
```
better to keep it compact.